src/serverenvironment.cpp

```diff
@@ -1256,6 +1256,8 @@ void ServerEnvironment::clearObjects(ClearObjectsMode mode)
 		obj->removingFromEnvironment();
 		// Deregister in scripting api
 		m_script->removeObjectReference(obj);
+		// stop attached sounds
+		m_server->stopAttachedSounds(id);
```

Comment:
clearobjects is now O(n²) if every object has a sound btw (but still I don't think there's necessarily any need to prematurely optimize this, since I don't expect thousands of objects or sounds to be around)

Reply:
I think it *should* be prematurely optimized. It's very easy too:
```cpp
void Server::stopAttachedSounds(const std::unordered_set<u16>&);

void Server::some_function() {
    std::unordered_set<u16> removed_ids;
    for (...) {
        // ...
        m_script->removeObjectReference(obj);
        removed_ids.insert(id);
        // ...
    }
    m_server->stopAttachedSounds(removed_ids);
}
```